### PR TITLE
Dynamic fetch aws contract abis

### PR DESCRIPTION
**Pull request by kangarang**
**What this does**
Dynamic fetching of smart contract ABIs. Also a little bit of BN.js

TODO:
- [X] getRegistry (fetch)
- [X] getToken (fetch)
- [x] getPLCRVoting (fetch)
- [x] getParameterizer (fetch)

### CODE REVIEW

**Comment by miguelmota** [comment]
great work so far on the refactoring

File: src/services/registry.js [hunk 1]
```diff
@@ -83,13 +85,13 @@ class RegistryService {
...
     try {
-      await token.approve(this.address, deposit)
+      await token.approve(this.address, bigDeposit)
     } catch (error) {
       throw error
     }
 
     try {
-      await this.registry.apply(domain, deposit, {from: this.getAccount()})
+      await this.registry.apply(domain, bigDeposit, {from: this.getAccount()})
```
Comment (miguelmota) [inline src/services/registry.js]: does it work without calling `<BN>.toNumber()` when passing as argument? @kangarang

Comment (kangarang) [reply src/services/registry.js]: @miguelmota it's supported, but we should always use `BigNumber`s for precision.

https://hackernoon.com/a-note-on-numbers-in-ethereum-and-javascript-3e6ac3b2fad9

Comment (miguelmota) [reply src/services/registry.js]:
```
Invalid number of arguments to Solidity function
```

Seems to be getting it confused with the options object. Works when using `toNumber()`

Comment (kangarang) [reply src/services/registry.js]: i see, yes, i'm getting the same error now. will report back soon

Comment (skmgoldin) [reply src/services/registry.js]: Try `toString()`. Should work, and safer than `toNumber()`.

Comment (skmgoldin) [reply src/services/registry.js]: Truffle-contract uses BigNumber, not BN, and they're usually interoperable but going to a string is pretty foolproof, and also lossless.

Comment (kangarang) [reply src/services/registry.js]: AHH that makes sense! thanks @skmgoldin

**Comment by kangarang** [comment]
Hi @miguelmota , I am finished with this PR, but no rush on the pull -- maybe it's best we wait til @eccheung4 implements dynamic ABI fetching on his backend

**Comment by skmgoldin** [comment]
Yeah probably makes sense to merge this after Edwin gets dynamic fetching working on the server.

**Pull request merged**